python/tank/authentication/sso_saml2/utils.py

```diff
@@ -46,18 +56,107 @@ def is_sso_enabled_on_site(shotgun_module, url, http_proxy=None):
         # require authentication.
         http_proxy = _sanitize_http_proxy(http_proxy).netloc
         if http_proxy:
-            get_logger().debug("Using HTTP proxy to connect to the Shotgun server: %s" % http_proxy)
-        info = shotgun_module.Shotgun(url, session_token="dummy", connect=False, http_proxy=http_proxy).info()
-        get_logger().debug("User authentication method for %s: %s" % (url, info["user_authentication_method"]))
-        if "user_authentication_method" in info:
-            return info["user_authentication_method"] == "saml2"
+            get_logger().debug(
+                "Using HTTP proxy to connect to the Shotgun server: %s", http_proxy
+            )
+        if url not in INFOS_CACHE or ((time.time() - INFOS_CACHE[url][0]) > INFOS_CACHE_TIMEOUT):
```

Comment:
Maybe add something like

`# Checks if the information in the cache is missing or out of date.`